### PR TITLE
CI: Add CI to test matplotlib against free-threaded Python

### PR DESCRIPTION
**Pull request by andfoy**
## PR summary

See https://github.com/matplotlib/matplotlib/issues/28611

This PR enables testing matplotlib against free-threaded Python, this is part of the ongoing effort to test projects against the changes proposed in [PEP703](https://peps.python.org/pep-0703/)

For more information see the following related PRs in other projects:
* NumPy https://github.com/numpy/numpy/pull/26512
* SciPy https://github.com/scipy/scipy/pull/20822
* PyWavelets https://github.com/PyWavelets/pywt/pull/753
* Pandas: https://github.com/pandas-dev/pandas/pull/59058
* Pillow: https://github.com/python-pillow/Pillow/pull/8200
* Scikit-image: https://github.com/scikit-image/scikit-image/pull/7463

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by andfoy** [comment]
Tests are passing now!

**Comment by tacaswell** [comment]
I do have a slight concern that this has significant copy-paste from the rest of our tests, but I also see an argument that this is easier to deal with than adding conditionals to the existing test ymls.

**Comment by andfoy** [comment]
@tacaswell I 100% agree with your assessment, however, as this new file could disappear once there are proper wheels for all the upstream dependencies (after Python 3.13 release), it can be used to set potential individual tweaks for future free-threaded releases.

**Comment by andfoy** [comment]
@tacaswell I think this PR is ready for review!

**Review by tacaswell — APPROVED** [review]
I am :+1: in principle, but would like @ksunden or @QuLogic to have the final review.

**Comment by tacaswell** [comment]
Can you please fix the linting issue ?

File: .github/workflows/free-threaded-tests.yml [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        steps:
+          - uses: actions/checkout@v4
+            with:
+              fetch-depth: 0
+
+          - name: Set up Python 3.13t
+            uses: deadsnakes/action@6c8b9b82fe0b4344f4b98f2775fcc395df45e494 # v3.1.0
+            with:
+                python-version: "3.13"
+                nogil: true
```
Comment (ksunden) [inline .github/workflows/free-threaded-tests.yml:56]:
```suggestion
            uses: deadsnakes/action@6c8b9b82fe0b4344f4b98f2775fcc395df45e494  # v3.1.0
            with:
              python-version: "3.13"
              nogil: true
```

In addition, the whole file should be dedented by 4 spaces (aside from the first line which has no spaces)

**Comment by QuLogic** [comment]
Thanks for the PR! I have rolled all these changes into #28689.

**Pull request closed without merging**